pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTMethodDeclaration.java

```diff
@@ -136,50 +151,25 @@ public boolean isVoid() {
 
 
     /**
-     * Returns the result type node of the method.
-     */
-    public ASTResultType getResultType() {
-        return getFirstChildOfType(ASTResultType.class);
-    }
-
-
-    /**
-     * Returns the block defined by this method, or
-     * null if the method is abstract.
+     * Returns the default clause, if this is an annotation method declaration
+     * that features one. Otherwise returns null.
      */
-    public ASTBlock getBlock() {
-        return getFirstChildOfType(ASTBlock.class);
+    @Nullable
+    public ASTDefaultValue getDefaultClause() {
+        return AstImplUtil.getChildAs(this, jjtGetNumChildren() - 1, ASTDefaultValue.class);
     }
 
-
     /**
-     * Returns the exception names listed in the {@code throws} clause
-     * of this method declaration, or null if there are none.
+     * Returns the result type node of the method.
      */
-    public ASTNameList getThrows() {
-        return getFirstChildOfType(ASTNameList.class);
+    public ASTResultType getResultType() {
+        return getFirstChildOfType(ASTResultType.class);
     }
 
-
     @Override
     public MethodLikeKind getKind() {
         return MethodLikeKind.METHOD;
     }
 
-    public ASTTypeParameters getTypeParameters() {
-        return getFirstChildOfType(ASTTypeParameters.class);
-    }
 
-    //@Override // enable this with PMD 7.0.0 - see interface ASTMethodOrConstructorDeclaration
-    public ASTFormalParameters getFormalParameters() {
-        return getFirstChildOfType(ASTMethodDeclarator.class).getFirstChildOfType(ASTFormalParameters.class);
-    }
-
-
-    /**
-     * Returns the method declarator. Never null.
-     */
-    public ASTMethodDeclarator getMethodDeclarator() {
```

Comment:
This method is not deprecated on master, but does it make sense to deprecate it?

Reply:
Hmm maybe for this one it does make sense. People can already use methods of ASTMethodDeclaration like getMethodName and getFormalParameters, and don't really need to use that node.

Reply:
Done in 04e5619fa884283a4fa024c871ca99b9f160d911